.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,64 @@
+name: "publish-npm"
+
+on:
+  release:
+    types: ["created"]
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          submodules: "recursive"
+
+      - name: "Install task"
+        run: "npm install -g @go-task/cli"
+
+      - name: "Run package task"
+        run: "task package"
+
+      - name: "Upload build artifacts"
+        uses: "actions/upload-artifact@v4"
+        with:
+          path: "dist/"
+
+  publish-npm:
+    needs: ["build"]
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+      - uses: "actions/download-artifact@v4"
+
+      - uses: "actions/setup-node@v4"
+        with:
+          node-version: 22
+          registry-url: "https://registry.npmjs.org/"
+
+      - run: "npm publish"
+        env:
+          NODE_AUTH_TOKEN: "${{secrets.NPM_TOKEN}}"
+
+  create-branch:
```

Comment:
Why do we need to create a branch?

Reply:
I thought that was a strategy used by our java packages (e.g., logback, log4j1-appenders).

With that said, since we cannot overpublish a version on npm, maybe we don't need to create a branch for any security bug fixes on a published version. What do you think?

Reply:
Ah, the branching strategy in the Java repos is just as a way to stage the release before tagging and releasing it. In those repos, we don't publish the release version update to the main branch, so it's easier to work with that change if it's on its own branch. Once the release is tagged, we eventually delete the branch.

Yeah, we'd never re-release a version afaik. If at all, we'd branch from a released version to create a new patch version.

Reply:
Gotcha. i'll remove this branching job then